SharpBucket/V2/RequestExecutorV2.cs

```diff
@@ -0,0 +1,24 @@
+﻿using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+using RestSharp;
+using SharpBucket.Authentication;
+
+namespace SharpBucket.V2
+{
+    internal class RequestExecutorV2 : RequestExecutor
+    {
+        protected override void AddBody(RestRequest request, object body)
+        {
+            // Use Newtonsoft.Json serialization instead of the one included in RestSharp to be able to
+            // ignore null properties during the serialization
+            // https://stackoverflow.com/questions/20006813/restsharp-how-to-skip-serializing-null-values-to-json
+            var jsonSettings = new JsonSerializerSettings
+            {
+                NullValueHandling = NullValueHandling.Ignore,
+                ContractResolver = new CamelCasePropertyNamesContractResolver()
+            };
+            var jsonString = JsonConvert.SerializeObject(body, jsonSettings);
```

Comment:
FYI I've write a test that call the AddNewBuildStatus, GetBuildStatusInfo and ChangeBuildStatusInfo in a branch on my copy here: 
https://github.com/mnivet/SharpBucket/tree/TestDeclinePullRequest_RemoveNewtonsoft

And I confirm that actual code is not working (conversion of the BuildInfoState enum is wrong if we do nothing)

* It can work if we serialize with Newtonsoft.Json like I've done here
* Or it can work if we serialize with the RestSharp serializer and with a customized JsonSerializerStrategy like I've done in the branch were I've write that tests

That second branch also demonstrate that we can totally drop the dependency to Newtonsoft.Json

Reply:
I've merge with FF the branch TestDeclinePullRequest_RemoveNewtonsoft into that pull request, then delete it, since it represent the second option on which we are agree